subsys/suit/stream/stream_sinks/src/suit_sdfw_sink.c

```diff
@@ -165,25 +120,60 @@ static suit_plat_err_t check_urot_none(const uint8_t *buf, size_t size)
 	return err;
 }
 
-static suit_plat_err_t check_urot_activated(const uint8_t *buf, size_t size)
+static suit_plat_err_t update_already_ongoing(const uint8_t *buf, size_t size)
 {
-	uint8_t *current_sdfw_digest = (uint8_t *)(NRF_SICR->UROT.SM.TBS.FW.DIGEST);
-	uint8_t *candidate_digest_in_manifest =
-		(uint8_t *)(buf + CONFIG_SUIT_SDFW_UPDATE_DIGEST_OFFSET);
+	suit_plat_err_t err = SUIT_PLAT_SUCCESS;
 
-	bool digests_match =
-		memcmp(candidate_digest_in_manifest, current_sdfw_digest,
-		       PSA_HASH_LENGTH(PSA_ALG_SHA_512))
-		       == 0;
+	enum sdfw_update_status update_status = sdfw_update_initial_status_get();
 
-	if (digests_match) {
-		LOG_DBG("Digest match - update success");
-		return SUIT_PLAT_SUCCESS;
+	/* Candidate is different than current FW but SDFW update is already ongoing. */
+	switch (update_status) {
+	case SDFW_UPDATE_STATUS_NONE: {
+		/* No pending operation even though operation indicates SDFW update.
+		 * Yet candidate differs from current FW, so schedule the update.
+		 * TODO: Check if this situation is possible or under what circumstances
+		 */
+		err = schedule_update_and_reboot(buf, size);
```

Comment:
I am not sure but shouldn't this result in an error if this situation seems so unlikely and we do not know what it means?

Reply:
Your approach is one of the possibilities - it is more restrictive, i.e. we do not attempt an update if situation is unusual. The other approach is to try the update, which still might be successful and let it fail elsewhere.
We can discuss this matter.

Reply:
Regarding your concern of reset-loop when scheduling the update during state with inconsistent/unusual register values:
- In both sinks, before each update schedule, there is an attempt to clear the registers. Failed clearing results in error, which should be processed by SUIT processor appropriately, leading to update termination or moving to the other slot. This means, inconsistent registers values should not be persistent
- in sdfw_update.c, after successful updates the registers are cleared
- in sdfw_update.c, when unexpected status is encountered, the registers are also cleared

As such, I believe it is unlikely to encounter a situation of reset loop caused by this fragment of code.